docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,169 @@
+- This design allows for grouping of metrics for which you would want to define particular customizations. Grouping allows hiveconfig to be parsable, but implementation of this model would require an efficient solution to validate there are no ambiguous entries for a metric, and storing it within the code for parsing purposes. We might end up enforcing only 1 entry for a metric in the HiveConfig.
+- We would be removing the fixed labels that can be propagated from the cluster deployment, i.e.`cluster_type`, `platform` and `version`. This means, if admins want these labels for the metrics, they will have to define them via  metricsConfig.metricsToReport[].additionalClusterDeploymentLabels. This would only be a breaking change if the admins suddenly cut over from deprecated/old way of metricsConfig to the new metricsToReport, sufficient documentation should help though.
+- We would be renaming `hive_cluster_deployment_install_failure_total` and `hive_cluster_deployment_install_success_total` to ensure it ends with `seconds`, as it is a duration-based metric.
+- We would no longer panic when the metricsConfig fails validation, instead we would be updating hiveConfig.Status.Conditions[ReadyCondition] to false, with appropriate reason and message. See [Failure Modes](#failure-modes) for more details.
+
+#### Deprecation
+
+In order to implement this change, we will have to deprecate the existing hiveconfig.spec.metricsConfig.metricsWithDuration and hiveconfig.spec.metricsConfig.additionalClusterDeploymentLabels.
+As a way to handle the deprecation, we will assert that in a hiveconfig, you can either provide the new configurations (aka `metricsToReport`) or the old way of hiveconfig.spec.metricsConfig.metricsWithDuration and hiveconfig.spec.metricsConfig.additionalClusterDeploymentLabels. If there's an empty metricsConfig, we would default to _old_ behaviour for the period of deprecation.
+Note that only the old design will support the mapping of camelCase key to metric for minimumDuration, the new way will require full metric names only.
+
+#### Failure modes
+
+These situations will result in metrics not getting registered and hiveConfig.Status.Conditions[ReadyCondition] will be set to false:.
+- metricsToReport.metricNames[].<name> doesn't exist as a metric
+- metricsToReport.metricNames[].<name> is a metric without duration but minimumDuration was specified.
+- a metric is defined across multiple groupings and the customizations defined contradict. For ex, following hiveConfig will cause panic because hive will be confused whether to parse `minimumDuration` for `hive_foo_seconds` to be 10m or 1h. Please note, we might end up allowing only 1 entry of hive metric in the hiveConfig.
```

Comment:
I hear your suggestion for allowing only 1 entry for a metric - I want to keep this avenue open and see how it pans out during implementation.

Reply:
As written you're suggesting that we would allow duplicates if they're effectively identical. What possible reason could there be for allowing that? It can only be confusing to the user looking at HiveConfig; and it makes our job harder in the code (how do you determine "effectively identical"?) So I say we take a hard stance here. Duplicates => error.

Reply:
Yeah, you're right. Let's not allow duplicate entries.